org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/preferences/internal/DefaultModelPersistence.java

```diff
@@ -54,32 +52,33 @@
 public final class DefaultModelPersistence implements ModelPersistence, EventListener {
 
     // Cache holding (project name - persistent model property) pairs.
```

Comment:
Commenting here, since I can't do that on your change request. 

There's one thing which worries me a bit.  Originally I wanted to avoid the null valules in the `PersistentModel` and in the `PersistentModelBuilder`. With your proposal, there'll be a bunch of null checks in the updaters, which seem a bit error-prone to me.